src/preamble.js

```diff
@@ -1032,7 +1032,14 @@ Module['asm'] = function(global, env, providedBuffer) {
     'element': 'anyfunc'
   });
   env['__memory_base'] = {{{ GLOBAL_BASE }}}; // tell the memory segments where to place themselves
+#if WASM_BACKEND
+  // We reserve slot 0 in the table for the NULL function pointer.
+  // This means the __table_base for the main module (even in dynamic linking)
+  // is alwasy 1.
+  env['__table_base'] = 1;
```

Comment:
this changes the non-relocatable case - do our non-relocatable wasm modules not depend on that value for placing their table segments? it seems like this would move them forward by 1, unless we never use this param in the non-relocatable case?

Reply:
__table_base and __memory_base (at least under the wasm backend) are ignored except for relocatable code.

I added a comment to clarify.